### ISSUE TITLE
deleting a key does not trigger effects 

### ISSUE THREAD

**Issue by bigmistqke** [luisherranz/deepsignal#21]
example:
```ts
const sign = deepSignal<{ key?: string }>({  });

setTimeout(() => {
  sign.key = 'value';
}, 1000);

setTimeout(() => {
  delete sign.key;
}, 2000);

effect(() => {
  console.log('sign', sign.key);
});
```

expected behavior:
1. log `undefined`
1. after 1 second log `value`
2. after 2 seconds log `undefined`

current behavior:
1. log `undefined`
1. after 1 second log `value`

[see](https://playground.solidjs.com/anonymous/ac0ef3cf-62eb-4f5f-9953-5e31821fd93a) for an example with solid's `createMutable`

**Comment by bigmistqke** [comment]
a fix in userland (and mb inspiration for implementation):

```ts
sign.key = undefined;
delete sign.key;
```

**Comment by luisherranz** [comment]
Thanks! I opened a PR to add support for `delete`: 

- https://github.com/luisherranz/deepsignal/pull/23

**Comment by luisherranz** [comment]
Released as [deepsignal@1.3.1](https://github.com/luisherranz/deepsignal/releases/tag/deepsignal%401.3.1).

@bigmistqke could you please test it out and see if everything works as expected now?

**Comment by bigmistqke** [comment]
@luisherranz awesome! thanks for notifying me. will let u know asap.
